src/activity/exe-unit/batch.ts

```diff
@@ -136,10 +135,10 @@ export class Batch {
     }
   }
 
-  async endStream(): Promise<Readable> {
+  async endStream(): Promise<Observable<Result>> {
     const script = this.script;
     await script.before();
-    let results: Readable;
+    let results: Observable<Result>;
     try {
       results = await this.executor.execute(this.script.getExeScriptRequest());
```

Comment:
Redundant await. At the same time that's no longer throwing as it's returing an Observable to she try/catch block here is also redundant.